### PR TITLE
New layout

### PR DESCRIPTION
**Pull request by raybellwaves**
new layout: moves tests to root folder, rm setup.py and use pyproject.toml
use latest s3fs moto patch from https://github.com/aio-libs/aiobotocore/issues/755#issuecomment-1424945194
added a KeyError(f"{urlpah}")
fixed a mypy complain with https://github.com/hauntsaninja/no_implicit_optional

### CODE REVIEW

**Comment by tdhopper** [comment]
you don't want to commit __pycache__ directories

File: .gitignore [hunk 1]
```diff
@@ -1,152 +0,0 @@
-!.cruft.json
-!.yamllint.yaml
-!.flake8
-!.mypy.ini
-!.editorconfig
-!.gitignore
```
Comment (blackary) [inline .gitignore:6]: Why did you delete this?

File: intake_pattern_catalog/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+from .catalog import PatternCatalog, PatternCatalogTransform  # noqa
+
+
+__version__ = "2023.2.0"
```
Comment (blackary) [inline intake_pattern_catalog/__init__.py:4]: RIP CalVer 😿

Comment (blackary) [reply intake_pattern_catalog/__init__.py:4]: Automagic CalVer, I should say :)

**Review by blackary — APPROVED** [review]
Seems like a fine set of changes overall. A few decisions that wouldn't be my favorite:
- I do like automagic calver, but doing it manually is certainly simpler
- I'd put the gitignore back
- I do prefer putting source code in `src/` to make it harder to accidentally import the local module, rather than the package, when testing the package

But, none of those seem like a big deal, and there are some significant wins here, so seems fine to me overall.

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-fsspec>=2021.4.0
+fsspec
 intake
-s3fs
-msgpack
+requests
```
Comment (blackary) [inline requirements.txt:3]: Any idea why we need this, when we don't use it explicitly in the codebase?

File: .gitignore (hunk 1 above)
Comment (raybellwaves) [reply .gitignore]: Got a bit too greedy of removing stuff. I agree this should stay and I see it's common in other repos e.g. https://github.com/pandas-dev/pandas/blob/main/.gitignore and https://github.com/tiangolo/fastapi/blob/master/.gitignore

File: requirements.txt (hunk 3 above)
Comment (raybellwaves) [reply requirements.txt:3]: https://github.com/intake/intake/issues/706
if Martin does another intake release we can remove

Comment (sfc-gh-zblackwood) [reply requirements.txt]: 🦸

**Comment by raybellwaves** [comment]
thanks for the review gents (@blackary @tdhopper). Going to merge and work on a release GHA
